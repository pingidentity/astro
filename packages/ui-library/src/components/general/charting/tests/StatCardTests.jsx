window.__DEV__ = true;

describe("StatCard", function () {
    const React = require("react"),
        ReactTestUtils = require("react-dom/test-utils"),
        TestUtils = require("../../../../testutil/TestUtils"),
        _ = require("underscore"),
        StatCard = require("../StatCard"),
        StatCardRow = require("../StatCardRow");

    const listData = [
        { label: "Last 30 days", value: "29" },
        { label: "Last 60 days", value: "124" },
        { label: "Last 90 days", value: "167" },
        { label: "Last 120 days", value: "195" },
        { label: "Last 150 days", value: "201" },
    ];

    function getComponent (props) {
        props = _.defaults(props || {}, {
            title: "Failed Attempts",
            description: "February 2016",
            value: "1,056",
            data: listData,
            accent: 1,
        });

        return ReactTestUtils.renderIntoDocument(<StatCard {...props} />);
    }

    it("data-id's don't change", () => {
        TestUtils.mountSnapshotDataIds(
            <StatCard
                data={listData}
                title="QUACK QUACK"
                value="0"
            />
        );
    });

    it("renders with default data-id", function () {
        const component = getComponent();
        const chart = TestUtils.findRenderedDOMNodeWithDataId(component, "stat-card");
        expect(chart).toBeTruthy();
    });

    it("renders a StatCardRow", function() {
        const component = ReactTestUtils.renderIntoDocument(<StatCardRow>content</StatCardRow>);

        expect(component).toBeDefined();
    });

    it("does not render the error or spinner by default", function () {
        const component = getComponent();

        const errorMessage = TestUtils.findRenderedDOMNodeWithClass(component, "stat-card__error");
        expect(errorMessage).toBeFalsy();

        const spinner = TestUtils.findRenderedDOMNodeWithClass(component, "stat-card__spinner");
        expect(spinner).toBeFalsy();
    });

    it("renders the list on back", function () {
        const component = ReactTestUtils.renderIntoDocument(<StatCardRow flipped={true}>content</StatCardRow>);
        const listMarkup = TestUtils.findRenderedDOMNodeWithClass(component, "dashboard-card__stat-list");

        expect(listMarkup).toBeDefined();
    });
});
