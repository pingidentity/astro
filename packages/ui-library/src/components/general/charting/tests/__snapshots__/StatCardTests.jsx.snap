// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard data-id's don't change 1`] = `
Array [
  "stat-card",
  "stat-card-list",
  "stat-card-view-toggle",
]
`;
